okio-zipfilesystem/src/jvmMain/kotlin/okio/zipfilesystem/ZipEntry.kt

```diff
@@ -38,13 +41,20 @@ internal class ZipEntry(
    *
    * At most 0xffff bytes when encoded.
    */
-  val name: String,
+  val canonicalPath: Path,
+
+  /**
+   * Determine whether or not this `ZipEntry` is a directory.
+   *
+   * @return `true` when this `ZipEntry` is a directory, `false` otherwise.
+   */
+  val isDirectory: Boolean = false,
 
   /**
    * Returns the comment for this `ZipEntry`, or `null` if there is no comment.
    * If we're reading a zip file using `ZipInputStream`, the comment is not available.
    */
-  val comment: String,
+  val comment: String = "",
```

Comment:
still non nullable while comment indicates null is no comment.

Reply:
Yeah, none of these comments make sense; they’re all still inherited from AOSP. I was hoping to go through all of them in a follow-up!